### ISSUE TITLE
The current provider doesn't support subscriptions: HttpProvider

### ISSUE THREAD

**Issue by mushketyk** [ConsenSys-archive/truffle#1633]
- [x] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Issue

I can't test events using Truffle 5.x. I get an error: `The current provider doesn't support subscriptions: HttpProvider`

## Steps to Reproduce

I have the following test code in a truffle project to check if an event was emitted:

```javascript
    it('event is emitted', async function(done) {
        let watcher = contract.Event(function(error, event) {
            console.log(`Error: ${error}`)
            console.log(`Event: ${event}`)
            
            done();
        });
       await triggerEvent();
    });
```

## Expected Behavior

`error` should be undefined, `event` should contain an emitted event.

## Actual Results

The code outputs the following output:

```
Error: Error: The current provider doesn't support subscriptions: HttpProvider
Event: undefined
```

## Environment

* Operating System: MacOS 10.14.2
* Ethereum client:
```
$ truffle version
Truffle v5.0.1 (core: 5.0.1)
Solidity v0.5.0 (solc-js)
Node v11.1.0

$ npm --version
6.4.1
```

**Comment by eggplantzzz** [comment]
Yes, so `HttpProvider` doesn't support subscriptions. I'm not sure what you are specifying in your Truffle config... Check out the web3 provider docs: https://web3js.readthedocs.io/en/1.0/web3-eth.html#providers

One option you have if you are trying to test events is to look at the logs. When you execute a method, the returned object has a `logs` property. Inspect this and you can see what events were emitted.

**Comment by mushketyk** [comment]
@eggplantzzz My Truffle config is very straightforward:

```javascript
module.exports = {
    networks: {
        ganache: {
            host: "localhost",
            port: 7545,
            gas: 5000000,
            network_id: "*"
        }
    }
};
```

It worked fine with testing events in Truffle 4.x, but does not work with Truffle 5.x. Do you know if anything was changed in relation to this?

I'll try to use the logs property instead.

**Comment by eggplantzzz** [comment]
Also @mushketyk, I'm not sure about the syntax you were using in 4...the web3 docs have a syntax like the following:
`myContract.once("myEvent"[, options], callback)...`
Check it out...
https://web3js.readthedocs.io/en/1.0/web3-eth-contract.html#once

**Comment by elenadimitrova** [comment]
@eggplantzzz `myContract.on("myEvent" ...` errors with `myContract.on is not a function`

Also if we enable `websockets` in the truffle config (cc @mushketyk) we instead get `Error: connection not open`. This error might be related to https://github.com/ethereum/web3.js/issues/1965 which was fixed in `beta.36` of `web3v1` and the latest truffle is one version behind that.

**Comment by eggplantzzz** [comment]
The latest Truffle uses Web3 version `1.0.0-beta.37`. I think Ganache might be using `35` maybe?
I'll check out the `websockets` setting.

**Comment by elenadimitrova** [comment]
Sorry I meant this is fixed in `web3-v1-beta.38`. Updated my comment above.

**Comment by emrem6** [comment]
any solution on that issue?

**Comment by eggplantzzz** [comment]
@emrem6 I think you should be able to use websockets with the latest Truffle (5.1.33). If you are still having a problem and can provide me with more detail I'll happily try and lend a hand.

You will need to supply a websockets provider and doctor up your network configuration a bit. See [this](https://github.com/trufflesuite/trufflesuite.com/pull/706) for some extra info.

**Comment by xi1570-krupeshanadkat** [comment]
I got the solution, notice the last line in config.

```js
    development: {
      host: "127.0.0.1",
      port: 9545,
      network_id: "*", // Match any network id
      websockets: true
    }
```

Register Event Watcher

```js
contract("Testing Flight Status Oracles", async (accounts) => {
  let instance;

  before('setup contract', async() => {
    // Initialize Config
    instance = await ExerciseC6D.deployed();

    // Monitor Events
    instance.Tester({}, (error, result) => {
      if(error) console.error(error);
      console.log(`[TESTER] => [message] : ${result.args.message} [status] : ${result.args.status}`);
    });
  })


  it("Should Register Oracles Successfully", async () => {
    await instance.registerOracle({from: accounts[0], value: 1})
  })
})
```

Solidity Event Declaration & Emitting
```sol
    // Event Tester
    event Tester(string message, bool status);
    function registerOracle() external payable {
      ...
      ...
      // Emit Event
      emit Tester("New Oracle Registered", true);
    }
```

I hope this helps!
